cmd/telegraf/cmd_plugins.go

```diff
@@ -0,0 +1,240 @@
+// Command handling for configuration "plugins" command
+package main
+
+import (
+	"fmt"
+	"io"
+	"sort"
+	"strings"
+
+	"github.com/influxdata/telegraf/plugins/aggregators"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/outputs"
+	"github.com/influxdata/telegraf/plugins/processors"
+	"github.com/influxdata/telegraf/plugins/secretstores"
+	"github.com/urfave/cli/v2"
+)
+
+func getInputNames() []byte {
+	inputNames := make([]string, 0, len(inputs.Inputs))
+	for i := range inputs.Inputs {
+		inputNames = append(inputNames, fmt.Sprintf("inputs.%s\n", i))
+	}
+	sort.Strings(inputNames)
+
+	return []byte(strings.Join(inputNames, ""))
+}
+
+func getOutputNames() []byte {
+	outputNames := make([]string, 0, len(outputs.Outputs))
+	for i := range outputs.Outputs {
+		outputNames = append(outputNames, fmt.Sprintf("outputs.%s\n", i))
+	}
+	sort.Strings(outputNames)
+
+	return []byte(strings.Join(outputNames, ""))
+}
```

Comment:
How about replacing all those functions with
```golang
func pluginNames[M ~map[string]V, V any](m M, prefix string) string {
	names := make([]string, 0, len(m))
	for k := range m {
		names = append(names, prefix+"."+k)
	}
	sort.Strings(names)
	return strings.Join(names, "\n")
}
```

and then call that with `pluginNames(outputs.Outputs, "outputs")` or the other types respectively? This would save quite some lines...